### PR TITLE
`SparseHamiltonian` uses `AnyWires

### PR DESCRIPTION
**Pull request by mudit2812**
Updated `SparseHamiltonian` to use `AnyWires` instead of `AllWires`.

### CODE REVIEW

**Comment by mudit2812** [comment]
[sc-34240]

**Review by albi3ro — COMMENTED** [review]
I'm surprised we don't need to change any tests.

Since our tests are completely independent of this change, we should add some tests. Since the `QNode` doesn't allow `AllWires` operators unless their wires match up with the device, maybe we should add an integration test where `SparseHamiltonian` only acts on some of the device wires.

File: pennylane/ops/qubit/observables.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ class SparseHamiltonian(Observable):
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/ops/qubit/observables.py:215]: Not true anymore.

**Review by albi3ro — APPROVED** [review]
:) glad we finally did that! Simple but just needed to get done.

**Pull request merged**